packages/csv-parser-discount-code/test/main.spec.js

```diff
@@ -1,5 +1,142 @@
-import multiply from '../src/main'
+// eslint-disable-next-line import/no-extraneous-dependencies
+import streamtest from 'streamtest'
+import fs from 'fs'
+import path from 'path'
+import CsvParser from '../src/main'
 
-it('should multiply 2 numbers', () => {
-  expect(multiply(3, 4)).toBe(12)
+describe('CsvParser', () => {
+  const logger = {
+    error: () => {},
+    warn: () => {},
+    info: () => {},
+    verbose: () => {},
+  }
+
+  let csvParser
+  beforeEach(() => {
+    csvParser = new CsvParser(logger)
+  })
+
+  describe('::constructor', () => {
+    it('should be a function', () => {
+      expect(typeof CsvParser).toBe('function')
+    })
+
+    it('should set default properties', () => {
+      expect(csvParser.delimiter).toBe(',')
+      expect(csvParser.multiValueDelimiter).toBe(';')
+      expect(csvParser.continueOnProblems).toBeFalsy()
+    })
+  })
+
+  describe('::_removeEmptyFields', () => {
+    it('should be defined', () => {
+      expect(CsvParser._removeEmptyFields).toBeDefined()
```

Comment:
This test is not needed because it will throw an error in the next test if the function is not there